### PR TITLE
Replace Guzzle 6 Adapter with Guzzle HTTP

### LINKED ISSUE #23: player()->get() throws a server error on player ID which not regtistered on TruckersMP
On https://github.com/TruckersMP/API-Client/blob/master/src/Requests/PlayerRequest.php#L56 method expects `response` parameter as array, but if player do not exist under that ID API returns `response` as a string.

### LINKED ISSUE #24: Use guzzlehttp/guzzle instead of php-http/guzzle6-adapter
If possible, could guzzlehttp/guzzle be used? This would be more consistent with other uses within the Laravel Framework. php-http/guzzle6-adapter is not required by https://github.com/TruckersMP/Steam-Socialite so I am unsure why it is needed here.

C:\laragon\www\plasmagaming>composer require truckersmp/api-client
Using version ^1.2 for truckersmp/api-client
./composer.json has been updated
Running composer update truckersmp/api-client
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - php-http/guzzle6-adapter[dev-master, v1.0.0, ..., v1.1.1, v2.0.0, ..., v2.0.2] require guzzlehttp/guzzle ^6.0 -> found guzzlehttp/guzzle[6.0.0, ..., 6.5.x-dev] but the package is fixed to 7.3.0 (lock file version) by a partial
 update and that version does not match. Make sure you list it as an argument for the update command.
    - truckersmp/api-client v1.2.0 requires php-http/guzzle6-adapter ^1.0 || ^2.0 -> satisfiable by php-http/guzzle6-adapter[v1.0.0, v1.1.0, v1.1.1, v2.0.0, v2.0.1, v2.0.2, 2.x-dev (alias of dev-master)].
    - php-http/guzzle6-adapter 2.x-dev is an alias of php-http/guzzle6-adapter dev-master and thus requires it to be installed too.
    - Root composer.json requires truckersmp/api-client ^1.2 -> satisfiable by truckersmp/api-client[v1.2.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

### PR DESCRIPTION
**Pull request by bensherred**
As the title suggests, this replaces the guzzle 6 adapter library with guzzle HTTP. This will solve issues when trying to use the package alongside the steam socialite package.

Closes #23 & #24

### CODE REVIEW

File: src/Requests/Request.php [hunk 1]
```diff
@@ -79,21 +59,41 @@ abstract public function get();
...
+
+    /**
+     * Check if the response contains an error.
+     *
+     * @param $response
+     * @return bool
+     */
+    protected function hasError($response): bool
+    {
+        if (! isset($response['error'])) {
```
Comment (ShawnCZek) [inline src/Requests/Request.php]: With the current styling, we do not put a space after an exclamation mark (e.g., the constructor of the Ban model).

File: composer.json [hunk 2]
```diff
@@ -44,10 +44,7 @@
     "require": {
         "php": "^7.3",
         "nesbot/carbon": "^2.16",
-        "ext-json": "*",
```
Comment (ShawnCZek) [inline composer.json:47]: What is the reason for removing `ext-json` from requirements?

**Review by ShawnCZek — COMMENTED** [review]
Everything seems to be alright apart from the two little issues I mention in the comments.
Once these issues are resolved, I see no reason why not to approve this request.

Although, because of the changes to exceptions, I believe this has to be a part of the next major version. Therefore, it would be great to look for breaking changes we might want to do in it as well.

File: composer.json (hunk 2 above)
Comment (bensherred) [reply composer.json:47]: No particular reason apart from personal preference. It's include required as part of `guzzlehttp/guzzle` so would be checked anyway.

**Review by ShawnCZek — APPROVED** [review]

**Pull request merged**